### PR TITLE
use a daemon thread to get water tiles instead

### PR DESCRIPTION
**Pull request by code-explorer786**
# **this pr is untested because i'm clearly on mobile**
please test this and see if the game crashes for whatever reason
thanks

# what is this all about
so i was playing on mobile with your mod (in a 500x500 map, unfortunately) and it suddenly gets lagspikes for some reason, and suddenly frogs appear
i checked the source code and found the cause, which is checking all tiles in the main thread
coincidentally, i found out about threads in Mindustry days ago, so why not

- i clearly did not test this code because i'm on mobile
- daemon threads because why not, they just get all water tiles and that's it
- thank you for your time reading this

### CODE REVIEW

**Comment by Sempyos** [comment]
A much good idea

File: scripts/units/ANIfrog.js [hunk 1]
```diff
@@ -247,18 +247,30 @@ frog.constructor = () => extend(MechUnit, {
 
 frog.defaultController = prov(() => extend(AIController, {}));
 
-Events.on(Trigger.update.getClass(), () => {
-    if(!Mathf.chanceDelta(0.0001) || !Vars.state.isPlaying()) return;
+var map_valids = [];
```
Comment (FlinTyX) [inline scripts/units/ANIfrog.js:250]: var is deprecated
let >

File: scripts/units/ANIfrog.js [hunk 2]
```diff
@@ -247,18 +247,30 @@ frog.constructor = () => extend(MechUnit, {
...
+var map_valids_usable = false:
```
Comment (FlinTyX) [inline scripts/units/ANIfrog.js:251]: `false:`
the only error of the entire code, but thats probably because of mobile keyboard

**Comment by FlinTyX** [comment]
The code works just fine.
Lag spikes are gone after the world is totaly loaded and the core lands.
I've made some changes to the old code, so I will merge it and then push the new one (it only has some team changes and thats it, your code will stay there)

**Pull request merged**